### PR TITLE
[testbed-cli] Enhance testbed-cli.sh script to support caching deployed topologies

### PR DESCRIPTION
**Pull request by shlomibitton**
Signed-off-by: Shlomi Bitton <shlomibi@nvidia.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?

Enhance testbed-cli.sh script to support caching deployed topologies.
This is to reduce time when a developer wants to deploy a certain topology on a testbed, so only cached topology will be removed instead of all known topologies.

#### How did you do it?

Add a caching mechanism, each time a user ask to add a topology using the testbed-cli script, it will be cached.
Once the user will ask to deploy a different topology using testbed-cli script, only the cached topology will be removed and the new topology will be configured.
The cached topologies will be saved on a shared location known to all sonic-mgmt dockers and servers.
The path of the shared location should be provided as a separate file named "cached_topologies_path" under the ansible directory.
This file should be provided by each vendor on the internal git.
This PR provide the infrastructure to have this functionality, if the file contains the path is missing, the script will abort.

#### How did you verify/test it?

Replace topologies on a testbed.

### CODE REVIEW

File: ansible/testbed-cli.sh [hunk 1]
```diff
@@ -550,6 +566,99 @@ function cleanup_vmhost
       --vault-password-file="${passwd}" -l "${server}" $@
 }
 
+function read_topologies_from_csv_file
+{
+  items_by_setup_name=$(cat $tbfile | grep $setup_name | awk 'BEGIN { FS = "," } ; {print $1}')
```
Comment (liorghub) [inline ansible/testbed-cli.sh:571]: pls change item to topology (for all places item* -> topology*)

File: ansible/testbed-cli.sh [hunk 2]
```diff
@@ -550,6 +566,99 @@ function cleanup_vmhost
...
+  for item in $items_by_setup_name
+  do
+      if [[ "$item" == *"$setup_name"* ]]; then
+          result_list+=("$item")
+      fi
+  done
+
+  echo ${result_list[@]}
```
Comment (liorghub) [inline ansible/testbed-cli.sh:579]: change result_list to topology_list

File: ansible/testbed-cli.sh [hunk 3]
```diff
@@ -550,6 +566,99 @@ function cleanup_vmhost
...
+}
+
+function get_setup_name_by_testbed_name
+{
+  testbed_line=$(cat $tbfile | grep -m 1 $testbed_name)
```
Comment (liorghub) [inline ansible/testbed-cli.sh:584]: Add remark with example

File: ansible/testbed-cli.sh [hunk 4]
```diff
@@ -259,6 +262,19 @@ function add_topo
   # Delete the obsoleted arp entry for the PTF IP
   ip neighbor flush $ptf_ip || true
 
+  cache_files_path="$(pwd)/cached_topologies_path"
```
Comment (liorghub) [inline ansible/testbed-cli.sh:265]: Add comment - what are you doing here?

File: ansible/testbed-cli.sh (hunk 2 above)
Comment (shlomibitton) [reply ansible/testbed-cli.sh:579]: Done

File: ansible/testbed-cli.sh (hunk 1 above)
Comment (shlomibitton) [reply ansible/testbed-cli.sh:571]: Done

File: ansible/testbed-cli.sh (hunk 3 above)
Comment (shlomibitton) [reply ansible/testbed-cli.sh:584]: No comments on this file except of the 'usage' part

File: ansible/testbed-cli.sh (hunk 4 above)
Comment (shlomibitton) [reply ansible/testbed-cli.sh]: Changed it to a function call and now it is more clear

**Review by roysr-nv — APPROVED** [review]

**Pull request closed without merging**